app/controllers/submissions_controller.rb

```diff
@@ -252,14 +258,33 @@ def score
     client_socket&.send_data(JSON.dump(@submission.calculate_score))
     # To enable hints when scoring a submission, uncomment the next line:
     # send_hints(client_socket, StructuredError.where(submission: @submission))
+
+    # Submit the score
+    if @submission.user.external_user? && lti_outcome_service?(@submission.exercise_id, @submission.user.id)
+      response = send_score(@submission)
+
+      if response[:status] == 'success'
+        if response[:score_sent] != @submission.normalized_score
+          # Score has been reduced due to the passed deadline
+          client_socket&.send_data(JSON.dump({cmd: :status, status: :scoring_too_late, score_sent: response[:score_sent]}))
+        end
+      else
+        client_socket&.send_data(JSON.dump({cmd: :status, status: :scoring_failure}))
+      end
+    end
+
+    if @submission.score == @submission.exercise.maximum_score
+      @url = finalize_submission_path(@submission)
+      client_socket&.send_data(JSON.dump({cmd: :status, status: :full_score_reached, url: @url}))
```

Comment:
I know that we have heavily reworked this part previously. Since the new architecture is already more flexible, we can probably rethink these lines and just "add" the part to send a JSON through the WebSocket connection there.

_This shouldn't be a dedicated comment, it's more a reminder about a nice architecture._